pg_shard--1.1.sql

```diff
@@ -140,3 +140,51 @@ $sync_table_metadata_to_citus$ LANGUAGE 'plpgsql';
 
 COMMENT ON FUNCTION sync_table_metadata_to_citus(text)
 		IS 'synchronize a distributed table''s pg_shard metadata to CitusDB';
+
+-- Prepares a specified distributed table to work with COPY operations by
+-- creating a temporary table and trigger to handle rows coming from a COPY.
+-- After each INSERT, a provided sequence is incremented to track the total -- number of copied rows.
+CREATE FUNCTION prepare_distributed_table_for_copy(relation regclass, sequence regclass) RETURNS void
+AS $pdtfc$
+	DECLARE
+		table_name text;
+		temp_table_name text;
+		attr_names text[];
+		attr_list text;
+		param_list text;
+		using_list text;
+		insert_command text;
+		func_tmpl CONSTANT text  := $$  CREATE FUNCTION pg_temp.copy_to_insert() RETURNS trigger
+										AS $cti$
+											BEGIN
+												EXECUTE %L USING %s;
+												PERFORM nextval(%L);
+												RETURN NULL;
+											END;
+										$cti$ LANGUAGE plpgsql VOLATILE;
+									$$;
+		table_tmpl CONSTANT text := 'CREATE TEMPORARY TABLE %I (LIKE %s)';
+		trg_tmpl CONSTANT text := $$CREATE TRIGGER copy_to_insert
```

Comment:
The func_tmpl has spaces between the $$ and the statement. Could we do something similar here?

Also, what do you think about expanding the names? I especially found trg_tmpl a bit hard to grok.